sim/core/item_swaps.go

```diff
@@ -256,23 +357,34 @@ func (swap *ItemSwap) swapWeapon(slot proto.ItemSlot) {
 }
 
 func (swap *ItemSwap) reset(sim *Simulation) {
-	if !swap.IsEnabled() || !swap.IsSwapped() {
+	if !swap.IsEnabled() {
 		return
 	}
+	if swap.IsSwapped() {
+		swap.SwapItems(sim, swap.slots)
```

Comment:
Hmmm you might be right.
However with your solution you could never Swap a full set since it would exclude weapons.

Would it not make more sense to just keep track of the original equipment/unequipped items and revert to the original state?

Reply:
@Polynomix I've added a `isReset` param and I am storing the initial equip/swapItems to always be able to restore these to defaults.